Stripe/STPAddressFieldTableViewCell.swift

```diff
@@ -0,0 +1,505 @@
+  }
+
+  func stateFieldCaption(forCountryCode countryCode: String?) -> String {
+    switch countryCode {
+    case "US":
+      return STPLocalizedString(
+        "State",
+        "Caption for State field on address form (only countries that use state , like United States)"
+      )
+    case "CA":
+      return STPLocalizedString(
+        "Province",
+        "Caption for Province field on address form (only countries that use province, like Canada)"
+      )
+    case "GB":
+      return STPLocalizedString(
+        "County",
+        "Caption for County field on address form (only countries that use county, like United Kingdom)"
+      )
+    default:
+      return STPLocalizedString(
+        "State / Province / Region",
+        "Caption for generalized state/province/region field on address form (not tied to a specific country's format)"
+      )
+    }
+  }
+
+  func placeholder(for addressFieldType: STPAddressFieldType) -> String {
+    switch addressFieldType {
+    case .name:
+      return STPLocalizationUtils.localizedNameString()
+    case .line1:
+      return STPLocalizedString("Address", "Caption for Address field on address form")
+    case .line2:
+      return STPLocalizedString(
+        "Apt.", "Caption for Apartment/Address line 2 field on address form")
+    case .city:
+      return STPLocalizedString("City", "Caption for City field on address form")
+    case .state:
+      return stateFieldCaption(forCountryCode: self.ourCountryCode)
+    case .zip:
+      return self.countryCodeIsUnitedStates
+        ? STPLocalizedString(
+          "ZIP Code",
+          "Caption for Zip Code field on address form (only shown when country is United States only)"
+        )
+        : STPLocalizedString(
+          "Postal Code",
+          "Caption for Postal Code field on address form (only shown in countries other than the United States)"
+        )
+    case .country:
+      return STPLocalizedString("Country", "Caption for Country field on address form")
+    case .email:
+      return STPLocalizationUtils.localizedEmailString()
+    case .phone:
+      return STPLocalizedString("Phone", "Caption for Phone field on address form")
+    }
+  }
+
+  func delegateCountryCodeDidChange(countryCode: String) {
+    if self.type == .country {
+      self.contents = countryCode
+    }
+
+    self.ourCountryCode = countryCode
+    self.updateTextFieldsAndCaptions()
+    self.setNeedsLayout()
+  }
+
+  func updateAppearance() {
+    self.backgroundColor = self.theme.secondaryBackgroundColor
+    self.contentView.backgroundColor = .clear
+    self.textField?.placeholderColor = theme.tertiaryForegroundColor
+    self.textField?.defaultColor = theme.primaryForegroundColor
+    self.textField?.errorColor = self.theme.errorColor
+    self.textField?.font = self.theme.font
+    self.setNeedsLayout()
+  }
+
+  var countryCodeIsUnitedStates: Bool {
+    self.ourCountryCode == "US"
+  }
+
+  public override func becomeFirstResponder() -> Bool {
```

Comment:
hmm, `@objc`?

Reply:
@objc is implied by @objc protocols, though we've been a little inconsistent about it